### PR TITLE
[#30] Implement Lombok plugin

### PR DESCRIPTION
**Pull request by semyonov-m-y**

### CODE REVIEW

**Comment by GetFire** [comment]
надо пофиксить pom.xml

**Review by holkwen — APPROVED** [review]

**Review by zozich — APPROVED** [review]
OK

File: src/main/java/net/greatstart/model/Investment.java [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
 
+import lombok.Data;
+
 import java.math.BigDecimal;
 import java.time.LocalDate;
 
+@Data
 public class Investment {
 
     private LocalDate dateOfInvestment;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/model/Investment.java:11]: MAJOR Remove this unused "dateOfInvestment" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/model/Project.java [hunk 2]
```diff
@@ -1,8 +1,11 @@
 package net.greatstart.model;
 
+import lombok.Data;
+
 import java.time.LocalDate;
 import java.util.List;
 
+@Data
 public class Project {
     private boolean isVerifed;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/model/Project.java:10]: MAJOR Remove this unused "isVerifed" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/model/Project.java [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
 
+import lombok.Data;
+
 import java.time.LocalDate;
 import java.util.List;
 
+@Data
 public class Project {
     private boolean isVerifed;
     private String name;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/model/Project.java:11]: MAJOR Remove this unused "name" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/model/ProjectDescription.java [hunk 4]
```diff
@@ -1,7 +1,10 @@
 package net.greatstart.model;
 
+import lombok.Data;
+
 import java.time.LocalDate;
 
+@Data
 public class ProjectDescription {
 
     private LocalDate date;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/model/ProjectDescription.java:10]: MAJOR Remove this unused "date" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/model/User.java [hunk 5]
```diff
@@ -1,7 +1,10 @@
 package net.greatstart.model;
 
+import lombok.Data;
+
 import java.util.List;
 
+@Data
 public class User {
     private int id;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/model/User.java:9]: MAJOR Remove this unused "id" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: src/main/java/net/greatstart/model/User.java [hunk 6]
```diff
@@ -1,7 +1,10 @@
 package net.greatstart.model;
 
+import lombok.Data;
+
 import java.util.List;
 
+@Data
 public class User {
     private int id;
     private String name;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/model/User.java:10]: MAJOR Remove this unused "name" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 24 issues
* ![MAJOR][MAJOR] 24 major

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Investment.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Investment.java#L12): Remove this unused "project" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Investment.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Investment.java#L13): Remove this unused "inv" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Investment.java#L14](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Investment.java#L14): Remove this unused "sum" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L12](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Project.java#L12): Remove this unused "desc" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L13](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Project.java#L13): Remove this unused "cost" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L14](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Project.java#L14): Remove this unused "minInvestment" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L15](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Project.java#L15): Remove this unused "addDate" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L16](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Project.java#L16): Remove this unused "owner" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [Project.java#L17](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/Project.java#L17): Remove this unused "investors" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [ProjectDescription.java#L11](https://github.com/Sheptytskyid/Startup_Platform/blob/3efd97e217f11f3203b42c46d9542f2ff1c2c71e/src/main/java/net/greatstart/model/ProjectDescription.java#L11): Remove this unused "description" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**